lib/main.dart

```diff
@@ -53,33 +56,75 @@ class RouteDetailsWidget extends StatelessWidget {
     return ReorderableListView(
       onReorder: route.move,
       children: [
-        for (var c in route.places)
-          RouteCellWidget(c),
+        for (var c in route.places) RouteCellWidget(c),
       ],
     );
   }
 }
 
+class MyScaffold extends StatefulWidget{
+  @override
+  _MyScaffoldState createState() => _MyScaffoldState();
+}
+
+class _MyScaffoldState extends State<MyScaffold>{
+  bool expanded = true;
+  var rightPanelWidth;
+  var buttonWidth;
+  @override
+  Widget build(BuildContext context) {
+    final mediaQuery = MediaQuery.of(context);
+    final width = mediaQuery.size.width;
+    rightPanelWidth = max(width / 4, 228.0);
+    buttonWidth = max(width / 20, 60.0);
+    final height = mediaQuery.size.height;
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Maps Sample App'),
+        backgroundColor: Colors.green[700],
+      ),
+      body: Stack(
+        alignment: Alignment.center,
+        children: [
+          Expanded(child: RouteWidget()),
+          Positioned(
+            right: 0,
+            top: 0,
+            height: height,
+            child: AnimatedContainer(
+              color: Colors.white,
+              duration: const Duration(milliseconds: 200),
+              width: expanded ? rightPanelWidth : 0,
+              child: RouteDetailsWidget(),
+            ),
+          ),
+          Positioned(
+            right: expanded ? rightPanelWidth - buttonWidth / 2 : 0,
+            width: buttonWidth,
+            child: FloatingActionButton(
+              child: Icon(
+                  expanded ? Icons.arrow_forward_ios : Icons.arrow_back_ios),
+              onPressed: () => setState(() {
+                expanded = !expanded;
+              }),
+            ),
+          ),
+        ],
+      ),
+      floatingActionButton: FloatingActionButton(
```

Comment:
remove FloatingActionButton here, please

Reply:
why

Reply:
Because it can't do anything useful

Reply:
kk